src/os/bluestore/BlueFS.cc

```diff
@@ -1971,9 +2013,54 @@ void BlueFS::_close_writer(FileWriter *h)
       bdev[i]->queue_reap_ioc(h->iocv[i]);
     }
   }
+
+  // only cache sst files; our purpose here is only to cache compacted
+  // files for a short period to warm the rocksdb caches.
+  if (h->writer_type == WRITER_SST) {
+    _write_cache_add(h->file);
+  }
+
   delete h;
 }
 
+void BlueFS::_write_cache_add(FileRef file)
+{
+  // add to write cache
+  write_cache_bytes += file->contents.length();
```

Comment:
Shouldn't we bypass adding to the cache if bluefs_write_cache_bytes == 0 ( or < file->contents.length())?